python/tutorials/03-matrix-multiplication.py

```diff
@@ -323,13 +348,17 @@ def matmul(a, b, activation=""):
 torch_output = torch.matmul(a, b)
 print(f"triton_output_with_fp16_inputs={triton_output}")
 print(f"torch_output_with_fp16_inputs={torch_output}")
-if torch.allclose(triton_output, torch_output, atol=1e-2, rtol=0):
+# Bigger tolerance for AMD MI200 devices.
+# MI200 devices use reduced precision fp16 and bf16 and flush input and
+# output denormal values to zero. Detailed info is at: https://pytorch.org/docs/stable/notes/numerical_accuracy.html#reduced-precision-fp16-and-bf16-gemms-and-convolutions-on-amd-instinct-mi200-devices
+rtol = 1e-2 if is_hip_mi200() else 0
+if torch.allclose(triton_output, torch_output, atol=1e-2, rtol=rtol):
     print("✅ Triton and Torch match")
 else:
     print("❌ Triton and Torch differ")
 
 TORCH_HAS_FP8 = hasattr(torch, "float8_e5m2")
-if TORCH_HAS_FP8:
+if TORCH_HAS_FP8 and not is_hip():
```

Comment:
maybe switch to explicitly check `is_cuda` given this is specific to cuda. (we don't have the third backend right now in tree so it might be a moot point; but explicitly checking cuda is more friendly to other backends in the future.)